### DISCUSSION TITLE
commit

### DISCUSSION THREAD

**Discussion by AnGys09** [INGCOM-UNRN-P1/cursada-2023#84] [Preguntas y respuestas] [answered]
hola estoy trabajando con visual, pero desde hace un rato que no puedo hacer commit, me sale en gris el boton , no se que puede ser :-s
![error commit](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/105929830/ec92f8eb-7df1-40bc-aeb8-424a1775014b)

**Comment by AnGys09** [comment]
ya lo resolvi toqueteando configuraciones de git ..... Lo habia resuelto momentaneamente pero sigue desactivo el boton de commit......

**Reply by martinvilu** [reply]
En la ubicación del repositorio, hacé `git status` en la terminal de git bash.

**Comment by martinvilu** [comment]
Viendo la notificación, ¿qué cambiaste, que te funcionó?, temporariamente

**Reply by AnGys09** [reply]
Si, no se que abre tocado, pero me deja hacer commit solo si antes escribo en consola  git add <proyecto>, tambien tenia un lindo Lio de carpetas por que tratando de corregir eso clone nuevamente e todo.... pufff en fin, al menos funciona con el  git add , gracias,

**Reply by martinvilu** [reply]
Pero por eso, ¿que fue lo que modificaste? ¿Que dice `git status`?

**Reply by martinvilu** [reply]
Aaaah, pera, creo que tengo una idea; fíjate si haces click derecho sobre la lista que está en la captura de pantalla.

Lo que describís, es el funcionamiento _normal_ de git, que por alguna razón antes no tenías que hacer.

1. haces cambios
2. los agregas para confirmar (`git add`)
3. los confirmas (`git commit`)
4. los subís a github (`git push`)

Si no hay nada que confirmar (paso 2, no tenes que 'confirmar', paso 3)

**Reply by AnGys09** [reply]
en realidad no se por que tengo como desactivado el boton de commit, pero cuando hago git status me dice esto:
```ps
PS C:\Users\angy_\GIT Programacion 2023\TP02\tp2-2023-AnGys09\output> git status
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   ../ejercicio7.c

no changes added to commit (use "git add" and/or "git commit -a")
PS C:\Users\angy_\GIT Programacion 2023\TP02\tp2-2023-AnGys09\output> 

```
 pero cuando pongo git add (ejercicio), me activa l boton de commit, me deja ponerle el nombre lo hago y despues sincronizo

**Reply by martinvilu** [reply]
Por eso, el add es parte del ciclo de trabajo, pero porque antes te funcionaba y ahora no solo puede explicarse con una actualización del visual studio code.

Mi recomendación/sugerencia es meterle a la consola para la mayor cantidad de operaciones posibles (no solo de git, sino también de compilación)

**Reply by martinvilu** [reply]
El botón de commit esta desactivado porque `Changes not staged for commit` (del status); el 'stage' es donde vas agregando cambios para confirmar

**Comment by AnGys09** [comment, marked as answer]
![imagen](https://github.com/INGCOM-UNRN-P1/cursada-2023/assets/105929830/b894db3a-e55a-4874-83b9-65d56e4b1fd1)
Bueno paso a informar, después de mucho probar me di cuenta que el problema estaba que estaba desactivado el STAGED CHANGES, y si esto no se ve o esta desactivado no te dja hacer los commits de la forma simple y comoda de siempre, lo aviso por si a alguien le pasa en algun momento. Gracias!!!
